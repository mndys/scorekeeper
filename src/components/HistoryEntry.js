import React from 'react'
import styled from 'styled-components'

export default function HistoryEntry({ nameOfGame, players }) {
  return (
    <Wrapper>
      <h2>{nameOfGame}</h2>
      {players.map((player, index) => (
        <Player key={index}>
          <span>{player.name}</span>
          <span>{player.score}</span>
        </Player>
      ))}
    </Wrapper>
  )
}

const Wrapper = styled.section`
  display: grid;
  gap: 10px;
  border-top: dotted 2px lightgrey;
`

const Player = styled.div`
  display: flex;
  justify-content: space-between;
`
