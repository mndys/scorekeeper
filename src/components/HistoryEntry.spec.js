import { render, screen } from '@testing-library/react'
import HistoryEntry from './HistoryEntry'

describe('HistoryEntry', () => {
  it('renders game data from props', () => {
    const { container } = render(
      <HistoryEntry
        nameOfGame="Foo"
        players={[
          { name: 'John', score: 10 },
          { name: 'Jane', score: 20 },
        ]}
      />
    )
    expect(screen.getByText('Foo')).toBeInTheDocument()
    expect(screen.getByText('John')).toBeInTheDocument()
    expect(screen.getByText('Jane')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
