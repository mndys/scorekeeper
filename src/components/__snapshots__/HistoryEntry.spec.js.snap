// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryEntry renders game data from props 1`] = `
.c0 {
  display: grid;
  gap: 10px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

<section
  class="c0"
>
  Foo
  <div
    class="c1"
  >
    <span>
      John
    </span>
    <span>
      10
    </span>
  </div>
  <div
    class="c1"
  >
    <span>
      Jane
    </span>
    <span>
      20
    </span>
  </div>
</section>
`;
